pkg/apis/core/validation/validation.go

```diff
@@ -806,10 +847,8 @@ func validateISCSIPersistentVolumeSource(iscsi *core.ISCSIPersistentVolumeSource
 	if (iscsi.DiscoveryCHAPAuth || iscsi.SessionCHAPAuth) && iscsi.SecretRef == nil {
 		allErrs = append(allErrs, field.Required(fldPath.Child("secretRef"), ""))
 	}
-	if iscsi.SecretRef != nil {
-		if len(iscsi.SecretRef.Name) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("secretRef", "name"), ""))
-		}
```

Comment:
I think I might be missing something here
When we say SecretReference don't we mean both namespace and name, I mean if either of these is invalid then the secret itself is un-usable and thus the respective volume controller will not be able to use it anyway
cc @jsafrane

Reply:
I agree a secret cannot be referred to if the name/namespace is missing from the secret reference, but the partial secret reference doesn't necessarily break the respective volume controller. 

The secret reference is an optional field in some PersistentVolumeSources. I think the first concern discussed in https://github.com/kubernetes/kubernetes/pull/103024#issuecomment-874848298 was if a PersistentVolumeSource tolerates a partial secret reference (e.g. treats it as nil) and doesn't return a fatal error 100% of the time, then we cannot tighten the validation; otherwise we may break existing user workflow when we upgrade